Cache/Hashing.lean

```diff
@@ -11,8 +11,27 @@ namespace Cache.Hashing
 
 open System IO
 
-/-- We cache the hash of each file for faster lookup -/
-abbrev HashM := StateT IO.HashMap IO
+structure HashMemo where
+  depsMap : Lean.HashMap FilePath (Array FilePath)
+  hashMap : HashMap
+  deriving Inhabited
+
+partial def insertDeps (hashMap : HashMap) (path : FilePath) (hashMemo : HashMemo) : HashMap :=
+  if hashMap.contains path then hashMap else
+  match (hashMemo.depsMap.find? path, hashMemo.hashMap.find? path) with
+  | (some deps, some hash) => deps.foldl (insertDeps · · hashMemo) (hashMap.insert path hash)
+  | _ => hashMap
+
+def HashMemo.filterByPatterns (hashMemo : HashMemo) (patterns : List String) : IO HashMap := do
```

Comment:
```suggestion
def HashMemo.filterByFileName (hashMemo : HashMemo) (fileNames : List String) : IO HashMap := do
```
since these aren't patterns any more